packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -149,22 +171,113 @@ library LibVerifying {
 
     function _markBlockVerified(
         TaikoData.State storage state,
+        TaikoData.Config memory config,
         TaikoData.Block storage blk,
         TaikoData.ForkChoice storage fc,
-        uint24 fcId,
-        address systemProver
+        uint24 fcId
     )
         private
     {
-        uint64 proofTime;
-        unchecked {
-            proofTime = uint64(fc.provenAt - blk.proposedAt);
+        TaikoData.Auction memory auction;
+        {
+            uint64 batchId = LibAuction.batchForBlock(config, blk.blockId);
+            auction = state.auctions[batchId % config.auctionRingBufferSize];
+
+            // For system prover, maybe the auction does not exist.
+            if (auction.batchId != batchId) {
+                assert(fc.prover == address(1));
+                auction.batchId = 0; // indicating auction yet to start
+            }
+        }
+
+        // Refund the diff to the proposer
+        assert(fc.gasUsed <= blk.gasLimit);
```

Comment:
the actually mined L2 block's gasLimit is `blk.gasLimit + LibL2Consts.ANCHOR_GAS_COST`, so `fc.gasUsed` may greater than `blk.gasLimit` here.